### ISSUE TITLE
Issues with fine-tuning: learning rate sensitivity and overfitting/divergence

### ISSUE THREAD

**Issue by Demonsnightmare** [naver/mast3r#144]
Hello, TKS For your great job.Now I’m encountering two stability issues while fine-tuning the mast3r-slam model and would appreciate some guidance.
**Issue 1: Extreme sensitivity to learning rate**
What happens: When using the recommended fine-tuning LR (1e-4), the model quickly “goes bad” (pose estimates diverge / reconstruction quality collapses). Only when I reduce the LR to 1e-6 or lower do I see expected fine-tuning behavior.And my dataset mainly includes about 1300 intensity images.
Question: Is this normal? Could it indicate a large domain gap between the pretrained weights and my data, requiring a much lower LR? Or should other hyperparameters (optimizer, weight decay, etc.) be adjusted accordingly?
**Issue 2: Early stopping needed to avoid divergence**
What happens:
1.Training loss seems to decrease normally at first, then plateaus and oscillates around a certain value.
2. If I train for more epochs (e.g., 5–6), the model eventually diverges (“goes bad”). If I stop early (epochs 2–3), results remain plausible.
The following pictures show the trend and the pic with fewer epoches succeeded in fine tuning the model while the other turned out to fail.  
<img width="4167" height="2951" alt="Image" src="https://github.com/user-attachments/assets/ccb35c58-742e-4963-91d9-2585546584cc" />
<img width="4167" height="2951" alt="Image" src="https://github.com/user-attachments/assets/265dcad9-7ead-4ee2-95d0-31eb367a469b" />
And the following is the training code I use now.

CUDA_VISIBLE_DEVICES=7 nohup torchrun --nproc_per_node=1 train.py \
    --train_dataset "1000 @ DEV(split='train', ROOT='/nas0/dataset/EF/train/seq002modify', aug_crop='auto', aug_monocular=0.005,resolution=[(512, 384), (512, 336), (512, 288), (512, 256), (512, 160)], n_corres=8192, nneg=0.5, transform=ColorJitter)" \
    --test_dataset "100 @ DEV(split='test', ROOT='/nas0/dataset/EF/train/seq002modify', resolution=(512,384), n_corres=1024, seed=777)" \
    --model "AsymmetricMASt3R(pos_embed='RoPE100', patch_embed_cls='ManyAR_PatchEmbed', img_size=(512, 512), head_type='catmlp+dpt', output_mode='pts3d+desc24', depth_mode=('exp', -inf, inf), conf_mode=('exp', 1, inf), enc_embed_dim=1024, enc_depth=24, enc_num_heads=16, dec_embed_dim=768, dec_depth=12, dec_num_heads=12, two_confs=True)" \
    --train_criterion "ConfLoss(Regr3D(L21, norm_mode='?avg_dis'), alpha=0.2) + 0.075*ConfMatchingLoss(MatchingLoss(InfoNCE(mode='proper', temperature=0.05), negatives_padding=0, blocksize=8192), alpha=10.0, confmode='mean')" \
    --test_criterion "Regr3D_ScaleShiftInv(L21, norm_mode='?avg_dis', gt_scale=True, sky_loss_value=0) + -1.*MatchingLoss(APLoss(nq='torch', fp=torch.float16), negatives_padding=12288)" \
    --pretrained "checkpoints/MASt3R_ViTLarge_BaseDecoder_512_catmlpdpt_metric.pth" \
    --lr 5e-7 --min_lr 1e-07 --warmup_epochs 1 --epochs 3 --batch_size 4 --accum_iter 2 \
    --save_freq 1 --keep_freq 5 --eval_freq 1 --disable_cudnn_benchmark \
    --output_dir "checkpoints/train26" > 0logtxt/train26.log 2>&1 &
